src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs

```diff
@@ -101,6 +110,82 @@ public void InitializeService(ServiceHost serviceHost)
         }
 
         /// <summary>
+        /// Handles a restore request
+        /// </summary>
+        internal async Task HandleRestorePlanRequest(
+            RestoreParams restoreParams,
+            RequestContext<RestorePlanResponse> requestContext)
+        {
+            RestorePlanResponse response = new RestorePlanResponse();
+            ConnectionInfo connInfo;
+            bool supported = IsBackupRestoreOperationSupported(restoreParams, out connInfo);
+
+            if (supported && connInfo != null)
+            {
+                RestoreDatabaseTaskDataObject restoreDataObject = this.restoreDatabaseService.CreateRestoreDatabaseTaskDataObject(restoreParams);
+                response = this.restoreDatabaseService.CreateRestorePlanResponse(restoreDataObject);
+            }
+            else
+            {
+                response.CanRestore = false;
+                response.ErrorMessage = "Restore is not supported"; //TOOD: have a better error message
+            }
+            await requestContext.SendResult(response);
+
+        }
+
+        /// <summary>
+        /// Handles a restore request
+        /// </summary>
+        internal async Task HandleRestoreRequest(
+            RestoreParams restoreParams,
+            RequestContext<RestoreResponse> requestContext)
+        {
+            RestoreResponse response = new RestoreResponse();
+            ConnectionInfo connInfo;
+            bool supported = IsBackupRestoreOperationSupported(restoreParams, out connInfo);
+
+            if (supported && connInfo != null)
+            {
+                try
+                {
+                    RestoreDatabaseTaskDataObject restoreDataObject = this.restoreDatabaseService.CreateRestoreDatabaseTaskDataObject(restoreParams);
+
+                    if (restoreDataObject != null)
+                    {
+                        // create task metadata
+                        TaskMetadata metadata = new TaskMetadata();
+                        metadata.ServerName = connInfo.ConnectionDetails.ServerName;
+                        metadata.DatabaseName = connInfo.ConnectionDetails.DatabaseName;
+                        metadata.Name = SR.Backup_TaskName;
+                        metadata.IsCancelable = true;
+                        metadata.Data = restoreDataObject;
+
+
+                        // create backup task and perform
+                        SqlTask sqlTask = SqlTaskManager.Instance.CreateTask(metadata, this.restoreDatabaseService.RestoreTaskAsync, restoreDatabaseService.CancelTaskAsync);
```

Comment:
>CreateTask [](start = 66, length = 10)

maybe we should have another method like RunTask, so the explicit Run() call isn't required?